templates/docs/examples/patterns/side-navigation/_toggle_script.js

```diff
@@ -6,16 +6,32 @@
 */
 function toggleDrawer(sideNavigation, show) {
   if (sideNavigation) {
+    const outsideDrawer = sideNavigation.querySelector('.p-side-navigation__toggle');
+    const insideDrawer = sideNavigation.querySelector('.p-side-navigation__toggle--in-drawer');
     if (show) {
       sideNavigation.classList.remove('is-collapsed');
       sideNavigation.classList.add('is-expanded');
+      insideDrawer.focus();
+      toggleTabindex(sideNavigation, false);
     } else {
       sideNavigation.classList.remove('is-expanded');
       sideNavigation.classList.add('is-collapsed');
+      outsideDrawer.focus();
+      toggleTabindex(sideNavigation, true);
     }
   }
 }
 
+/**
+  Ensures links are only focusable when side nav is in view
+  @param {HTMLElement} sideNavigation The side navigation element.
+  @param {Boolean} sideNavCollapsed Whether the side nav is expanded or collapsed.
+*/
+function toggleTabindex(sideNavigation, sideNavCollapsed) {
+  const links = sideNavigation.querySelectorAll('.p-side-navigation__link');
```

Comment:
If we are doing this, I wonder if we can just completely remove the side nav content from the document?
I mean 'detach' it from DOM when it is collapsed (hidden) and attach again when it is expanded. But I guess it may conflict with toggle animation.

Reply:
I guess that's the same problem as if we would set `display: none` in CSS. We don't do that, because we animate the closing of the sidebar, so we can only remove it after the animation is finished, but this is handled by CSS.